src/manager/Manager.sol

```diff
@@ -78,71 +57,58 @@ contract Manager is IManager, VersionedContract, UUPS, Ownable, ManagerStorageV1
 
     /// @notice Deploys a DAO with custom token, auction, and governance settings
     /// @param _founderParams The DAO founders
-    /// @param _tokenParams The ERC-721 token settings
-    /// @param _auctionParams The auction settings
-    /// @param _govParams The governance settings
+    /// @param _implAddresses The implementation addresses
+    /// @param _implData The encoded list of implementation data
     function deploy(
         FounderParams[] calldata _founderParams,
-        TokenParams calldata _tokenParams,
-        AuctionParams calldata _auctionParams,
-        GovParams calldata _govParams
-    )
-        external
-        returns (
-            address token,
-            address metadata,
-            address auction,
-            address treasury,
-            address governor
-        )
-    {
+        address[] calldata _implAddresses,
+        bytes[] calldata _implData
+    ) external returns (address token, address metadata, address auction, address treasury, address governor) {
         // Used to store the address of the first (or only) founder
         // This founder is responsible for adding token artwork and launching the first auction -- they're also free to transfer this responsiblity
         address founder;
 
         // Ensure at least one founder is provided
         if ((founder = _founderParams[0].wallet) == address(0)) revert FOUNDER_REQUIRED();
 
+        uint256 implAddressesLength = _implAddresses.length;
+
+        // Ensure implementation parameters are correct length
+        if (implAddressesLength != IMPLEMENTATION_TYPE_COUNT || _implData.length != IMPLEMENTATION_TYPE_COUNT) revert INVALID_IMPLEMENTATION_PARAMS();
+
+        // Ensure all implementations are registered
+        unchecked {
+            for (uint256 i; i < implAddressesLength; ++i) {
+                if (!isImplementation[uint8(i)][_implAddresses[i]]) revert IMPLEMENTATION_NOT_REGISTERED();
+            }
+        }
+
         // Deploy the DAO's ERC-721 governance token
-        token = address(new ERC1967Proxy(tokenImpl, ""));
+        token = address(new ERC1967Proxy(_implAddresses[IMPLEMENTATION_TYPE_TOKEN], ""));
 
         // Use the token address to precompute the DAO's remaining addresses
         bytes32 salt = bytes32(uint256(uint160(token)) << 96);
 
         // Deploy the remaining DAO contracts
-        metadata = address(new ERC1967Proxy{ salt: salt }(metadataImpl, ""));
-        auction = address(new ERC1967Proxy{ salt: salt }(auctionImpl, ""));
-        treasury = address(new ERC1967Proxy{ salt: salt }(treasuryImpl, ""));
-        governor = address(new ERC1967Proxy{ salt: salt }(governorImpl, ""));
+        metadata = address(new ERC1967Proxy{ salt: salt }(_implAddresses[IMPLEMENTATION_TYPE_METADATA], ""));
```

Comment:
salt isn't really needed but i don't think it hurts this – since our impl addresses now can change